lib/mlutil.js

```diff
@@ -155,6 +155,12 @@ function endpointTransform(transform) {
 function databaseParam(connectionParams, endpoint, separator) {
   var database = connectionParams.database;
   if (database == null) {
+    if(connectionParams.basePath) {
+      let basePath = connectionParams.basePath.toString().startsWith('/') ? connectionParams.basePath.toString().substring(1):connectionParams.basePath;
```

Comment:
This code will be more readable if you first do:

    const basePath = connectionParams.basePath.toString();

And then to make it clear that the user's original base path isn't going to be altered (which makes code easier to understand), I would do:

    let fixedBasePath = basePath.startsWith("/") ? basePath.substring(1) : basePath;
    fixedBasePath = fixedBasedPath.endsWith("/") ? fixedBasedPath.substring(0, fixedBasePath.length - 1) : fixedBasePath;
    endpoint = fixedBasedPath + endpoint;